src/plots/declarative.py

```diff
@@ -361,21 +459,64 @@ def _set_yscale(self, ax, yscale):
         """
         ax.set_yscale(yscale)
 
+    def _sharex(self, ax):
+        """
+        If sharex axis is True, will find where to hide xticklabels.
+        """
+        if not ax.is_last_row():
+            plt.setp(ax.get_xticklabels(), visible=False)
+
+    def _sharey(self, ax):
+        """
+        If sharey axis is True, will find where to hide yticklabels.
+        """
+        if not ax.is_first_col():
+            plt.setp(ax.get_yticklabels(), visible=False)
+
+    def _add_map_features(self, ax, map_features):
+        """
+        Factory to add map features.
+        """
+        feature_dict = {
+            'coastline': cfeature.COASTLINE,
+            'borders': cfeature.BORDERS,
+            'states': cfeature.STATES,
+            'lakes': cfeature.LAKES,
+            'rivers': cfeature.RIVERS,
+            'land': cfeature.LAND,
+            'ocean': cfeature.OCEAN
+        }
+
+        for feat in map_features:
+            try:
+                ax.add_feature(feature_dict[feat])
+            except KeyError:
+                raise TypeError(f'{feat} is not a valid map feature.' +
+                                'Current map features supported are:\n' +
+                                f'{" | ".join(feature_dict.keys())}"')
+
 
 class CreatePlot():
     """
     Creates a figure to plot data as a scatter plot,
     histogram, or line plot.
     """
-
-    def __init__(self, plot_layers=[]):
+    def __init__(self, plot_layers=[], projection=None,
+                 domain=None):
 
         self.plot_layers = plot_layers
 
+        ###############################################
+        # Need a better way of doing this
+        if projection is not None and domain is not None:
+            self.projection = projection
+            self.domain = domain
+        ###############################################
```

Comment:
@CoryMartin-NOAA what are your thoughts on this? It is essential for determining if the object in the loop in `create_figure()` is a map or standard plot. Is there a better way?

Reply:
I think it's okay for now